### PR TITLE
Drops "type" from "script", "link" and "style" tags

### PR DESCRIPTION
**Pull request by newbazz**
Closes https://github.com/plone/Products.CMFPlone/issues/1078

### CODE REVIEW

File: Products/CMFPlone/static/components/ace-builds/kitchen-sink.html [hunk 1]
```diff
@@ -11,10 +11,10 @@
       commit  
     -->
 
-    <link rel="stylesheet" href="demo/kitchen-sink/styles.css" type="text/css" media="screen" charset="utf-8">
+    <link rel="stylesheet" href="demo/kitchen-sink/styles.css" media="screen" charset="utf-8">
     <script async="true" src="https://use.edgefonts.net/source-code-pro.js"></script>
 
-    <link href="./doc/site/images/favicon.ico" rel="icon" type="image/x-icon">
```
Comment (hvelarde) [inline Products/CMFPlone/static/components/ace-builds/kitchen-sink.html]: you can't remove the type attribute for the favicon.

File: Products/CMFPlone/static/tinymce-AtD-plugin/support/unignore.html [hunk 2]
```diff
@@ -5,8 +5,8 @@
 
 <p>Select a phrase to unignore it:</p>
 
-<SCRIPT TYPE="text/javascript" SRC="atdphrases.js"></SCRIPT>
```
Comment (hvelarde) [inline Products/CMFPlone/static/tinymce-AtD-plugin/support/unignore.html]: I think this script is no longer used and must be removed form the code on a different pull request; let's ask the @plone/framework-team.

**Comment by hvelarde** [comment]
@newbazz you need to add an entry on the changelog also.

File: Products/CMFPlone/static/components/ace-builds/kitchen-sink.html (hunk 1 above)
Comment (newbazz) [reply Products/CMFPlone/static/components/ace-builds/kitchen-sink.html]: ooh why is it so? it is also in the <link> tag.

**Comment by newbazz** [comment]
@hvelarde what is a changelog entry, i am not aware about that.

**Comment by hvelarde** [comment]
is a line in this file explaining the changes you made:

https://github.com/plone/Products.CMFPlone/blob/master/CHANGES.rst

in this case you have to add it in the bug fixes; somehting like "Remove deprecated `type` attributes from `script` and `style` tags" should be fine.

File: Products/CMFPlone/static/components/ace-builds/kitchen-sink.html (hunk 1 above)
Comment (hvelarde) [reply Products/CMFPlone/static/components/ace-builds/kitchen-sink.html]: because this is a different `link` type:

https://developer.mozilla.org/en-US/docs/Web/HTML/Link_types

**Comment by newbazz** [comment]
okk got you, i will make the given changes once i become the member of plone org so that all test pass all together.

**Comment by thet** [comment]
@newbazz tnx for the contribution, but there is a problem with it: all files (except ``Products/CMFPlone/static/patterns/toolbar/toolbar-demo.html``) are 3rd party libraries and will be automatically replaced when running bower next time. There is not much point of replacing them. Also most files you did the replacement are just demo or test files, so there is no effect with removing the type attribute anyways.

**Comment by newbazz** [comment]
@thet ok so what should i do?

**Comment by thet** [comment]
I'll close the issue, wait for the license agreement to be accepted and submit PRs for some addons in https://github.com/collective/ in the meantinme.

Regarding the issue of removing the type attribute - that has to be done in CMFPlone's resource registry itself.

**Comment by thet** [comment]
@newbazz the type attribute is most probably set somewhere in here: https://github.com/plone/Products.CMFPlone/blob/master/Products/CMFPlone/resources/

**Comment by newbazz** [comment]
@thet the url u gave contains no html files?

**Comment by hvelarde** [comment]
@thet would you mind to give a little more information to @newbazz; it could be discouraging just to close the PR.

**Comment by thet** [comment]
sorry, you're right - that could have felt a bit discouraging. wasn't meant so.

@newbazz - here - in these page templates - are the type attributes set:
https://github.com/plone/Products.CMFPlone/blob/master/Products/CMFPlone/resources/browser/styles.pt
https://github.com/plone/Products.CMFPlone/blob/master/Products/CMFPlone/resources/browser/scripts.pt

please don't forget to add a changelog entry to:

these documents can help jump-starting into plone development:
coredev contribution guidelines - https://docs.plone.org/develop/coredev/docs/index.html
coding style guidelines: https://docs.plone.org/develop/styleguide/index.html

happy coding!

**Comment by thet** [comment]
@newbazz do you want me to reopen the PR and you do a force-push onto your branch or do you just reopen another pull-request?

**Comment by newbazz** [comment]
Yes you can reopen this y to open a new PR.

**Comment by newbazz** [comment]
@thet changes made.

**Comment by jensens** [comment]
just FYI: ``master`` branch targets future Plone 5.2 (it will take while until there will be  a release). 

Since this is a considered as a bugfix it should be back-ported to the 5.1 branch, which will be the next release (current state: 5.1-pending release).

File: Products/CMFPlone/resources/browser/scripts.pt [hunk 3]
```diff
@@ -10,11 +10,11 @@
 </tal:if><tal:if condition="condcomment">
     <tal:opencc tal:replace="structure string:&lt;!--[if ${condcomment}]&gt;" />
 </tal:if>
-  <script type="text/javascript" tal:attributes="src script/src; data-bundle script/bundle"></script>
+  <script tal:attributes="src script/src; data-bundle script/bundle"></script>
 <tal:if condition="condcomment">
   <tal:closecc tal:condition="condcomment" tal:replace="structure string:&lt;![endif]--&gt;" />
 </tal:if><tal:if condition="resetrjs">
-  <tal:openredefine content='structure string:&lt;script type="text/javascript"&gt;'/>
+  <tal:openredefine content='structure string:&lt;script &gt;'/>
```
Comment (hvelarde) [inline Products/CMFPlone/resources/browser/scripts.pt]: please remove the space between `script` and `&gt;`.

File: Products/CMFPlone/resources/browser/scripts.pt [hunk 4]
```diff
@@ -1,6 +1,6 @@
-<script type="text/javascript" tal:content="string:PORTAL_URL = '${view/site_url}';"></script>
+<script tal:content="string:PORTAL_URL = '${view/site_url}';"></script>
 <tal:scripts repeat="script view/scripts"><tal:block define="condcomment script/conditionalcomment; resetrjs script/resetrjs|nothing"><tal:if condition="resetrjs">
-  <tal:openreset content='structure string:&lt;script type="text/javascript"&gt;'/>
+  <tal:openreset content='structure string:&lt;script &gt;'/>
```
Comment (hvelarde) [inline Products/CMFPlone/resources/browser/scripts.pt]: please remove the space between `script` and `&gt;`.

File: CHANGES.rst [hunk 5]
```diff
@@ -76,3 +76,5 @@ Bug Fixes:
   Completly removed ``came_from`` on ``@@register`` link.
   It does not make much sense anyway and we test nowhere if there is a came_from on that link.
   [jensens]
+
+- Removed depricated ``type`` attribute from ``style`` ``script`` and ``link`` tags.
```
Comment (hvelarde) [inline CHANGES.rst]: Remove deprecated `type` attribute from `script` and `link` tags.

**Comment by newbazz** [comment]
@hvelarde changes made.

**Comment by hvelarde** [comment]
@newbazz do you mind to cherry pcik these changes on the 5.1 branch in another pull request?

**Comment by newbazz** [comment]
@hvelarde actually i dont know how to do that.

**Comment by newbazz** [comment]
image
@hvelarde is this what u r asking for?

**Comment by hvelarde** [comment]
in your fork you have to checkout the 5.1 branch; cherry-pcik will not work because the changelog is going to be different, so you have to modify the same files:

```console
$ git checkout 5.1
$ git cherry-pick 3fb2e9c  <-- this probably will not work
```

**Comment by newbazz** [comment]
@hvelarde modify manually?

**Comment by hvelarde** [comment]
my mistake, branch name is 5.1.x.

**Comment by newbazz** [comment]
@hvelarde made changes in that branch and committed, now?

**Comment by hvelarde** [comment]
now just wait until the pull requests are merged; you can also look for more tasks.

File: CHANGES.rst [hunk 6]
```diff
@@ -76,3 +76,5 @@ Bug Fixes:
   Completly removed ``came_from`` on ``@@register`` link.
   It does not make much sense anyway and we test nowhere if there is a came_from on that link.
   [jensens]
+
+- Remove depricated ``type`` attribute from ``script`` and ``link`` tags.
```
Comment (hvelarde) [inline CHANGES.rst]: I forgot to say that you need to add your name below this entry: `[newbazz]`

Comment (hvelarde) [inline CHANGES.rst]: I forgot to say that you need to add your name below this entry: `[newbazz]`
